.github/workflows/e2e.yaml

```diff
@@ -14,18 +14,18 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
 
     - name: Cache node modules
```

Comment:
Now that you're having the caching from `actions/setup-node` you can remove this entire step.

Reply:
@kklamberty did you merge without removing this?

Reply:
Yes. I'm not confident that you are right (or wrong) about this step being unneceassary. The node cache step specifically says it does not cache node_modules, and this step aims to do that. We are not sure if it's working or not working, but it seems that it's _different_ in the least. 

This step does not cache node_modules, as is noted here: https://github.com/actions/setup-node
Different sources have varying opinions on whether or not we *should* cache node_modules, but since our step name suggests that we were trying to do so, I didn't think it would be problematic to keep our attempt in here.

In fact - @NicMcPhee and @jpwalbran - I think that looking at what to do if we have a cache hit might do the trick.
https://github.com/actions/cache#skipping-steps-based-on-cache-hit - I don't think we were ever looking for the cache and behaving differently based on whether or not we had a cache hit.

Reply:
This step _isn't_ caching the `node_modules` directory (which we shouldn't be). It sets up caching just for `~/.npm` (see the `path` below) which is the same as what setup-node is doing.

Reply:
`~/.npm` is a directory npm uses specifically for caching. We should not cache `node_modules` as that wouldn't work with `npm ci` and could break when the node version changes between runs.